patterns/1-initial/secure-innersource.md

```diff
@@ -0,0 +1,144 @@
+
+- Employee training about security awareness and individual responsibility
+- Enhanced security measures on source code management system such as:
+    - Eliminate anonymous access to source code (all users must be logged in to your Source Code Management tool), log the access to repositories
+    - regular review by organization owners on created repositories and granted permission to organization members.
+    - Set up [Single Sign-On][single-sign-on] against your organization’s identity provider. This will automatically revoke access in case somebody leaves the organization.
+    - regular rotation of access token
+    - 2FA authentication enforced
+    - encourage the use of signed commits
+- Monitoring the source code management system for suspicious access e.g.:
+    - access from new accounts or new devices
+    - downloading source code from new devices
+    - downloading a great amount of source code in the monitoring period.
+
+### Prevent sensitive information in shared repositories
+
+Build up agreed security requirement of InnerSource like:
+
+- No hard-coded usernames and/or passwords.
+- Repo does not contain client information.
+- Business critical algorithms may not be placed in SHARED repositories.
+- Code from SHARED repositories will not be distributed directly to Production.
+- Leveraging secrets scanning tools to scan the target repositories (including code, test cases, and helper scripts) for confidential data such as accounts, passwords, access tokens, keys, and other sensitive data.
+- Keep repositories CLOSED, but expose metadata about the projects (i.e. through an [InnerSource Portal][innersource-portal]), and create some kind of access request workflow. This way you could still give people access to the code, but not open it to everyone by default. *(Note: The full model is described under **Extension: An Additional Sharing Level**)*
+
+### Extension: An Additional Sharing Level
+
+In some cases, introducing additional sharing levels might be appropriate. Use cases include semi-sensitive code that can be shared with larger groups of people, but for example not with all interns and contractors. Or not with people with certain nationalities or working in a specific country, in order to comply with [export control][export-control] regulations. Or legacy repositories that are not (yet) ready to be made available as INTERNAL. Without the additional level, (too) many repositories will end up as CLOSED in an existing organization, as all businesses will considering their code 'special', requiring extra protection and/or 'not ready for opening up company wide yet (e.g. hard-coded secrets, but also technical debt, lacking documentation).
+
+And given 'findability' is also important, adding the additional level that facilitates findability, but still provides a light-weight gate seems to be a good way out to steer projects out of closed source.
```

Comment:
```suggestion
And given findability is also important, adding the additional level that facilitates findability, but still provides a light-weight access gate could be an appropriate way to prevent too many CLOSED repositories.
```